### PR TITLE
Ian/group upload

### PR DESCRIPTION
**Pull request by ianjorgensen**
I had issues testing because of the new postBlockToPlatform function in the uploader, but finally confirmed that it all works. Next up is having errors from /data in jellyfish show up in the chrome uploader and prompting the user to try again. Ready for review. @jh-bate

### CODE REVIEW

**Comment by ianjorgensen** [comment]
@jh-bate can you ping me when you get to look at this

**Comment by kentquirk** [comment]
@ianjorgensen, please merge master into this. It's quite old at this point.

**Comment by ianjorgensen** [comment]
@kentquirk thanks, i just merged and re tested

**Comment by kentquirk** [comment]
Wow, that's a MUCH easier PR to review. :) I did a little bit of cleanup, commenting, and typo fixes -- so @jh-bate if you could give it a once-over, I'd appreciate it.

File: app.js [hunk 1]
```diff
@@ -174,18 +183,49 @@ var jsonp = function(response) {
...
+              for (var id in groups) {
+                var group = groups[id];
+
+                if (id === req.params.groupId && (group.upload || group.root)) {
+                  return cb(null, id);
+                }
+              }
+
+              cb({
+                statusCode: 500,
```
Comment (jh-bate) [inline app.js]: hmm - not really a 500 is it?

File: app.js [hunk 2]
```diff
@@ -174,18 +183,49 @@ var jsonp = function(response) {
...
+                message: 'You don\'t have rights to upload to that account.'
+              });
+            });
+          },
+          function(id, cb) {
```
Comment (jh-bate) [inline app.js]: just checking but is this a groupId or userId? just so it reads consistently with the others

**Comment by jh-bate** [comment]
cool - added a couple of comments to be looked at first

File: app.js (hunk 1 above)
Comment (kentquirk) [inline app.js]: Good catch -- I changed it to a 403.

File: app.js (hunk 2 above)
Comment (kentquirk) [inline app.js]: Cleaned this up so that the roles of all the variables are clearer.

**Comment by kentquirk** [comment]
I'm happy with this now -- @jh-bate ?

**Comment by jh-bate** [comment]
yeah LGTM now

**Comment by jh-bate** [comment]
all up-to-date and have tested locally with existing uploader

**Comment by kentquirk** [comment]
Great stuff. Thanks.

**Pull request merged**